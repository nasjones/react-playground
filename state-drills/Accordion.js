import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: [
            {
                title: 'Section 1',
                content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
                index: 1
            },
            {
                title: 'Section 2',
                content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
                index: 2
            },
            {
                title: 'Section 3',
                content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
                index: 3
            },
        ]
    };

    state = {
        activeP: 0
    }

    renderButtons() {
        return this.props.sections.map((section) => (
            <li key={section.index}>
                <button onClick={() => this.handleButtonClick(section.index)}>{section.title}</button>
                <br />
                {this.renderContent(section)}
            </li>
        ))
    }

    handleButtonClick(change) {
        this.setState({
            activeP: change
        })
    }

    renderContent(section) {
        let output = ""
        if (section.index === this.state.activeP)
            output = section.content
        return output
    }
    render() {
        return (
            <div>
                <p>Accordion goes here</p>
                <ul>
                    {this.renderButtons()}
                </ul>
            </div>
        )
    }
}
export default Accordion