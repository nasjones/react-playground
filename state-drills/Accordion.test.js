import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe(`Accordion Component`, () => {
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders the no sections by default', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('opens any clicked section', () => {
        const wrapper = shallow(<Accordion />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only opens one section at a time', () => {
        const wrapper = shallow(<Accordion />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})