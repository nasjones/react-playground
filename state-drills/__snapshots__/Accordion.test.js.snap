// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<div>
  <p>
    Accordion goes here
  </p>
  <ul>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <br />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <br />
    </li>
    <li
      key="3"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <br />
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </li>
  </ul>
</div>
`;

exports[`Accordion Component opens any clicked section 1`] = `
<div>
  <p>
    Accordion goes here
  </p>
  <ul>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <br />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <br />
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </li>
    <li
      key="3"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <br />
    </li>
  </ul>
</div>
`;

exports[`Accordion Component renders the no sections by default 1`] = `
<div>
  <p>
    Accordion goes here
  </p>
  <ul>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <br />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <br />
    </li>
    <li
      key="3"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <br />
    </li>
  </ul>
</div>
`;
